server/container_create_linux.go

```diff
@@ -411,6 +414,27 @@ func (s *Server) mountImage(ctx context.Context, specgen *generate.Generator, im
 
 	log.Infof(ctx, "Image mounted to: %s", mountPoint)
 
+	if m.ImageSubPath != "" {
+		// Kubernetes already verifies that ImageSubPath:
+		// - is a relative path
+		// - does not contain any '..'
+		//
+		// We cannot create non-existing paths within the image volume because
+		// they're mounted read-only.
+		mountPointFile, err := securejoin.OpenInRoot(mountPoint, m.ImageSubPath)
```

Comment:
when will it be closed?

Reply:
Ah, I forgot that good catch. Defer closing right after opening should be fine from my point of view.